### ISSUE TITLE
Problem with installation

### ISSUE THREAD

**Issue by FionaMoon** [aristoteleo/dynamo-release#327]
Hi Dr.Qiu,
I met problem when I try to install `dynamo` in py3.9 environment using `pip install dynamo-release/ --user`.
It seems that something wrong with dependencies, but I don' know how to fix it.
Error shows as following:
```
$ pip install dynamo-release/ --user
Processing ./dynamo-release
  Installing build dependencies ... error
  error: subprocess-exited-with-error
  
  × pip subprocess to install build dependencies did not run successfully.
  │ exit code: -9
  ╰─> [20 lines of output]
      Collecting flit
        Using cached flit-3.7.1-py3-none-any.whl (49 kB)
      Collecting requests
        Using cached requests-2.27.1-py2.py3-none-any.whl (63 kB)
      Collecting tomli-w
        Using cached tomli_w-1.0.0-py3-none-any.whl (6.0 kB)
      Collecting flit_core>=3.7.1
        Using cached flit_core-3.7.1-py3-none-any.whl (60 kB)
      Collecting docutils
        Using cached docutils-0.18.1-py2.py3-none-any.whl (570 kB)
      Collecting tomli
        Using cached tomli-2.0.1-py3-none-any.whl (12 kB)
      Collecting urllib3<1.27,>=1.21.1
        Using cached urllib3-1.26.9-py2.py3-none-any.whl (138 kB)
      Collecting charset-normalizer~=2.0.0
        Using cached charset_normalizer-2.0.12-py3-none-any.whl (39 kB)
      Collecting certifi>=2017.4.17
        Using cached certifi-2021.10.8-py2.py3-none-any.whl (149 kB)
      Collecting idna<4,>=2.5
        Using cached idna-3.3-py3-none-any.whl (61 kB)
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: subprocess-exited-with-error

× pip subprocess to install build dependencies did not run successfully.
│ exit code: -9
╰─> See above for output.
```
Thank you inadvance!
Fiona

**Comment by Xiaojieqiu** [comment]
HI @FionaMoon thanks for using our tool! can you install with the second approach mentioned in our readthedocs page? 

```
git clone https://github.com/aristoteleo/dynamo-release.git
cd dynamo-release/
python setup.py install --user
```

We are releasing a new dynamo today or tomorrow. and will take a look at your issues too

**Comment by FionaMoon** [comment]
Hi ,
I've tried install from git download. But some strange things happen. The installation stops and shows that 
> Finished processing dependencies for dynamo-release==1.0.9
Then I try to import dynamo and meet the error 
> Package metadata: could not find distribution “dynamo”.

**Comment by Xiaojieqiu** [comment]
Could you please let us know the full error message? I cannot figure out what is going on with just excerpts of those info. Thanks! 

Meanwhile I just released a new version to PyPi (dynamo-release 1.0.9), https://pypi.org/project/dynamo-release/

You can also try to pip install? This will give you the most latest release which is basically what we have on github.

**Comment by FionaMoon** [comment]
Sorry for not getting back to you sooner.
I re-install by ` pip install dynamo-release`, and get info as following:
```
Requirement already satisfied: dynamo-release in /home/liangyue/.local/lib/python3.9/site-packages/dynamo_release-1.0.9-py3.9.egg (1.0.9)
Requirement already satisfied: numpy>=1.20.0 in ./miniconda/envs/py39/lib/python3.9/site-packages (from dynamo-release) (1.21.5)
Requirement already satisfied: pandas>=1.1.2 in ./miniconda/envs/py39/lib/python3.9/site-packages (from dynamo-release) (1.4.0)
Requirement already satisfied: scipy>=1.4.1 in ./miniconda/envs/py39/lib/python3.9/site-packages (from dynamo-release) (1.8.0)
Requirement already satisfied: scikit-learn>=0.19.1 in ./miniconda/envs/py39/lib/python3.9/site-packages (from dynamo-release) (1.0.2)
Requirement already satisfied: cvxopt>=1.2.3 in /home/liangyue/.local/lib/python3.9/site-packages/cvxopt-1.3.0-py3.9-linux-x86_64.egg (from dynamo-release) (1.3.0)
Requirement already satisfied: anndata==0.7.5 in /home/liangyue/.local/lib/python3.9/site-packages/anndata-0.7.5-py3.9.egg (from dynamo-release) (0.7.5)
Requirement already satisfied: loompy>=3.0.5 in ./miniconda/envs/py39/lib/python3.9/site-packages (from dynamo-release) (3.0.6)
Requirement already satisfied: matplotlib>=3.4.1 in ./miniconda/envs/py39/lib/python3.9/site-packages (from dynamo-release) (3.5.1)
Requirement already satisfied: setuptools in ./miniconda/envs/py39/lib/python3.9/site-packages (from dynamo-release) (60.8.1)
Requirement already satisfied: numdifftools>=0.9.39 in /home/liangyue/.local/lib/python3.9/site-packages/numdifftools-0.9.40-py3.9.egg (from dynamo-release) (0.9.40)
Requirement already satisfied: umap-learn>=0.5.1 in ./miniconda/envs/py39/lib/python3.9/site-packages (from dynamo-release) (0.5.2)
Requirement already satisfied: PATSY>=0.5.1 in ./miniconda/envs/py39/lib/python3.9/site-packages (from dynamo-release) (0.5.2)
Requirement already satisfied: statsmodels>=0.9.0 in ./miniconda/envs/py39/lib/python3.9/site-packages (from dynamo-release) (0.13.2)
Requirement already satisfied: numba>=0.54.0 in ./miniconda/envs/py39/lib/python3.9/site-packages (from dynamo-release) (0.55.1)
Requirement already satisfied: seaborn>=0.9.0 in ./miniconda/envs/py39/lib/python3.9/site-packages (from dynamo-release) (0.11.2)
Requirement already satisfied: colorcet>=2.0.1 in /home/liangyue/.local/lib/python3.9/site-packages/colorcet-3.0.0-py3.9.egg (from dynamo-release) (3.0.0)
Requirement already satisfied: tqdm in ./miniconda/envs/py39/lib/python3.9/site-packages (from dynamo-release) (4.62.3)
Requirement already satisfied: python-igraph>=0.7.1 in /home/liangyue/.local/lib/python3.9/site-packages/python_igraph-0.9.9-py3.9.egg (from dynamo-release) (0.9.9)
Requirement already satisfied: pynndescent>=0.5.2 in ./miniconda/envs/py39/lib/python3.9/site-packages (from dynamo-release) (0.5.6)
Requirement already satisfied: pre-commit in /home/liangyue/.local/lib/python3.9/site-packages/pre_commit-2.18.1-py3.9.egg (from dynamo-release) (2.18.1)
Requirement already satisfied: networkx in ./miniconda/envs/py39/lib/python3.9/site-packages (from dynamo-release) (2.6.3)
Requirement already satisfied: nxviz==0.7.3 in /home/liangyue/.local/lib/python3.9/site-packages/nxviz-0.7.3-py3.9.egg (from dynamo-release) (0.7.3)
Requirement already satisfied: get_version>=3.5.4 in /home/liangyue/.local/lib/python3.9/site-packages/get_version-3.5.4-py3.9.egg (from dynamo-release) (3.5.4)
Requirement already satisfied: openpyxl in /home/liangyue/.local/lib/python3.9/site-packages/openpyxl-3.0.9-py3.9.egg (from dynamo-release) (3.0.9)
Requirement already satisfied: h5py in ./miniconda/envs/py39/lib/python3.9/site-packages (from anndata==0.7.5->dynamo-release) (3.6.0)
Requirement already satisfied: natsort in ./miniconda/envs/py39/lib/python3.9/site-packages (from anndata==0.7.5->dynamo-release) (8.1.0)
Requirement already satisfied: packaging in ./miniconda/envs/py39/lib/python3.9/site-packages (from anndata==0.7.5->dynamo-release) (21.3)
Requirement already satisfied: more-itertools>=8.6.0 in ./miniconda/envs/py39/lib/python3.9/site-packages (from nxviz==0.7.3->dynamo-release) (8.12.0)
Requirement already satisfied: palettable>=3.3.0 in ./miniconda/envs/py39/lib/python3.9/site-packages (from nxviz==0.7.3->dynamo-release) (3.3.0)
Requirement already satisfied: mkdocs in ./miniconda/envs/py39/lib/python3.9/site-packages (from nxviz==0.7.3->dynamo-release) (1.3.0)
Requirement already satisfied: mkdocs-material in ./miniconda/envs/py39/lib/python3.9/site-packages (from nxviz==0.7.3->dynamo-release) (8.2.8)
Requirement already satisfied: mknotebooks in /home/liangyue/.local/lib/python3.9/site-packages/mknotebooks-0.7.1.post0.dev497-py3.9.egg (from nxviz==0.7.3->dynamo-release) (0.7.1.post0.dev497)
Requirement already satisfied: param>=1.7.0 in ./miniconda/envs/py39/lib/python3.9/site-packages (from colorcet>=2.0.1->dynamo-release) (1.12.1)
Requirement already satisfied: pyct>=0.4.4 in ./miniconda/envs/py39/lib/python3.9/site-packages (from colorcet>=2.0.1->dynamo-release) (0.4.8)
Requirement already satisfied: dunamai in /home/liangyue/.local/lib/python3.9/site-packages/dunamai-1.11.1-py3.9.egg (from get_version>=3.5.4->dynamo-release) (1.11.1)
Requirement already satisfied: click in ./miniconda/envs/py39/lib/python3.9/site-packages (from loompy>=3.0.5->dynamo-release) (8.0.3)
Requirement already satisfied: numpy-groupies in ./miniconda/envs/py39/lib/python3.9/site-packages (from loompy>=3.0.5->dynamo-release) (0.9.14)
Requirement already satisfied: kiwisolver>=1.0.1 in ./miniconda/envs/py39/lib/python3.9/site-packages (from matplotlib>=3.4.1->dynamo-release) (1.3.2)
Requirement already satisfied: cycler>=0.10 in ./miniconda/envs/py39/lib/python3.9/site-packages (from matplotlib>=3.4.1->dynamo-release) (0.11.0)
Requirement already satisfied: pyparsing>=2.2.1 in ./miniconda/envs/py39/lib/python3.9/site-packages (from matplotlib>=3.4.1->dynamo-release) (3.0.7)
Requirement already satisfied: fonttools>=4.22.0 in ./miniconda/envs/py39/lib/python3.9/site-packages (from matplotlib>=3.4.1->dynamo-release) (4.29.1)
Requirement already satisfied: python-dateutil>=2.7 in ./miniconda/envs/py39/lib/python3.9/site-packages (from matplotlib>=3.4.1->dynamo-release) (2.8.2)
Requirement already satisfied: pillow>=6.2.0 in ./miniconda/envs/py39/lib/python3.9/site-packages (from matplotlib>=3.4.1->dynamo-release) (9.0.1)
Requirement already satisfied: llvmlite<0.39,>=0.38.0rc1 in ./miniconda/envs/py39/lib/python3.9/site-packages (from numba>=0.54.0->dynamo-release) (0.38.0)
Requirement already satisfied: algopy>=0.4 in ./miniconda/envs/py39/lib/python3.9/site-packages (from numdifftools>=0.9.39->dynamo-release) (0.5.7)
Requirement already satisfied: pytz>=2020.1 in ./miniconda/envs/py39/lib/python3.9/site-packages (from pandas>=1.1.2->dynamo-release) (2021.3)
Requirement already satisfied: six in ./miniconda/envs/py39/lib/python3.9/site-packages (from PATSY>=0.5.1->dynamo-release) (1.16.0)
Requirement already satisfied: joblib>=0.11 in ./miniconda/envs/py39/lib/python3.9/site-packages (from pynndescent>=0.5.2->dynamo-release) (1.1.0)
Requirement already satisfied: igraph==0.9.9 in ./miniconda/envs/py39/lib/python3.9/site-packages (from python-igraph>=0.7.1->dynamo-release) (0.9.9)
Requirement already satisfied: texttable>=1.6.2 in ./miniconda/envs/py39/lib/python3.9/site-packages (from igraph==0.9.9->python-igraph>=0.7.1->dynamo-release) (1.6.4)
Requirement already satisfied: threadpoolctl>=2.0.0 in ./miniconda/envs/py39/lib/python3.9/site-packages (from scikit-learn>=0.19.1->dynamo-release) (3.1.0)
Requirement already satisfied: et-xmlfile in /home/liangyue/.local/lib/python3.9/site-packages/et_xmlfile-1.1.0-py3.9.egg (from openpyxl->dynamo-release) (1.1.0)
Requirement already satisfied: cfgv>=2.0.0 in ./miniconda/envs/py39/lib/python3.9/site-packages (from pre-commit->dynamo-release) (3.3.1)
Requirement already satisfied: identify>=1.0.0 in ./miniconda/envs/py39/lib/python3.9/site-packages (from pre-commit->dynamo-release) (2.4.12)
Requirement already satisfied: nodeenv>=0.11.1 in ./miniconda/envs/py39/lib/python3.9/site-packages (from pre-commit->dynamo-release) (1.6.0)
Requirement already satisfied: pyyaml>=5.1 in ./miniconda/envs/py39/lib/python3.9/site-packages (from pre-commit->dynamo-release) (6.0)
Requirement already satisfied: toml in ./miniconda/envs/py39/lib/python3.9/site-packages (from pre-commit->dynamo-release) (0.10.2)
Requirement already satisfied: virtualenv>=20.0.8 in ./miniconda/envs/py39/lib/python3.9/site-packages (from pre-commit->dynamo-release) (20.14.0)
Requirement already satisfied: distlib<1,>=0.3.1 in ./miniconda/envs/py39/lib/python3.9/site-packages (from virtualenv>=20.0.8->pre-commit->dynamo-release) (0.3.4)
Requirement already satisfied: filelock<4,>=3.2 in ./miniconda/envs/py39/lib/python3.9/site-packages (from virtualenv>=20.0.8->pre-commit->dynamo-release) (3.6.0)
Requirement already satisfied: platformdirs<3,>=2 in ./miniconda/envs/py39/lib/python3.9/site-packages (from virtualenv>=20.0.8->pre-commit->dynamo-release) (2.4.1)
Requirement already satisfied: Jinja2>=2.10.2 in ./miniconda/envs/py39/lib/python3.9/site-packages (from mkdocs->nxviz==0.7.3->dynamo-release) (3.0.3)
Requirement already satisfied: Markdown>=3.2.1 in ./miniconda/envs/py39/lib/python3.9/site-packages (from mkdocs->nxviz==0.7.3->dynamo-release) (3.3.6)
Requirement already satisfied: importlib-metadata>=4.3 in ./miniconda/envs/py39/lib/python3.9/site-packages (from mkdocs->nxviz==0.7.3->dynamo-release) (4.11.1)
Requirement already satisfied: mergedeep>=1.3.4 in ./miniconda/envs/py39/lib/python3.9/site-packages (from mkdocs->nxviz==0.7.3->dynamo-release) (1.3.4)
Requirement already satisfied: pyyaml-env-tag>=0.1 in ./miniconda/envs/py39/lib/python3.9/site-packages (from mkdocs->nxviz==0.7.3->dynamo-release) (0.1)
Requirement already satisfied: ghp-import>=1.0 in ./miniconda/envs/py39/lib/python3.9/site-packages (from mkdocs->nxviz==0.7.3->dynamo-release) (2.0.2)
Requirement already satisfied: watchdog>=2.0 in ./miniconda/envs/py39/lib/python3.9/site-packages (from mkdocs->nxviz==0.7.3->dynamo-release) (2.1.7)
Requirement already satisfied: pygments>=2.10 in ./miniconda/envs/py39/lib/python3.9/site-packages (from mkdocs-material->nxviz==0.7.3->dynamo-release) (2.11.2)
Requirement already satisfied: mkdocs-material-extensions>=1.0.3 in ./miniconda/envs/py39/lib/python3.9/site-packages (from mkdocs-material->nxviz==0.7.3->dynamo-release) (1.0.3)
Requirement already satisfied: pymdown-extensions>=9.0 in ./miniconda/envs/py39/lib/python3.9/site-packages (from mkdocs-material->nxviz==0.7.3->dynamo-release) (9.3)
Requirement already satisfied: nbconvert>=6.0.0 in ./miniconda/envs/py39/lib/python3.9/site-packages (from mknotebooks->nxviz==0.7.3->dynamo-release) (6.4.1)
Requirement already satisfied: jupyter-client in ./miniconda/envs/py39/lib/python3.9/site-packages (from mknotebooks->nxviz==0.7.3->dynamo-release) (7.1.2)
Requirement already satisfied: gitpython in ./miniconda/envs/py39/lib/python3.9/site-packages (from mknotebooks->nxviz==0.7.3->dynamo-release) (3.1.27)
Requirement already satisfied: zipp>=0.5 in ./miniconda/envs/py39/lib/python3.9/site-packages (from importlib-metadata>=4.3->mkdocs->nxviz==0.7.3->dynamo-release) (3.7.0)
Requirement already satisfied: MarkupSafe>=2.0 in ./miniconda/envs/py39/lib/python3.9/site-packages (from Jinja2>=2.10.2->mkdocs->nxviz==0.7.3->dynamo-release) (2.0.1)
Requirement already satisfied: nbformat>=4.4 in ./miniconda/envs/py39/lib/python3.9/site-packages (from nbconvert>=6.0.0->mknotebooks->nxviz==0.7.3->dynamo-release) (5.1.3)
Requirement already satisfied: jupyter-core in ./miniconda/envs/py39/lib/python3.9/site-packages (from nbconvert>=6.0.0->mknotebooks->nxviz==0.7.3->dynamo-release) (4.9.1)
Requirement already satisfied: mistune<2,>=0.8.1 in ./miniconda/envs/py39/lib/python3.9/site-packages (from nbconvert>=6.0.0->mknotebooks->nxviz==0.7.3->dynamo-release) (0.8.4)
Requirement already satisfied: jupyterlab-pygments in ./miniconda/envs/py39/lib/python3.9/site-packages (from nbconvert>=6.0.0->mknotebooks->nxviz==0.7.3->dynamo-release) (0.1.2)
Requirement already satisfied: pandocfilters>=1.4.1 in ./miniconda/envs/py39/lib/python3.9/site-packages (from nbconvert>=6.0.0->mknotebooks->nxviz==0.7.3->dynamo-release) (1.5.0)
Requirement already satisfied: nbclient<0.6.0,>=0.5.0 in ./miniconda/envs/py39/lib/python3.9/site-packages (from nbconvert>=6.0.0->mknotebooks->nxviz==0.7.3->dynamo-release) (0.5.10)
Requirement already satisfied: defusedxml in ./miniconda/envs/py39/lib/python3.9/site-packages (from nbconvert>=6.0.0->mknotebooks->nxviz==0.7.3->dynamo-release) (0.7.1)
Requirement already satisfied: entrypoints>=0.2.2 in ./miniconda/envs/py39/lib/python3.9/site-packages (from nbconvert>=6.0.0->mknotebooks->nxviz==0.7.3->dynamo-release) (0.4)
Requirement already satisfied: bleach in ./miniconda/envs/py39/lib/python3.9/site-packages (from nbconvert>=6.0.0->mknotebooks->nxviz==0.7.3->dynamo-release) (4.1.0)
Requirement already satisfied: traitlets>=5.0 in ./miniconda/envs/py39/lib/python3.9/site-packages (from nbconvert>=6.0.0->mknotebooks->nxviz==0.7.3->dynamo-release) (5.1.1)
Requirement already satisfied: testpath in ./miniconda/envs/py39/lib/python3.9/site-packages (from nbconvert>=6.0.0->mknotebooks->nxviz==0.7.3->dynamo-release) (0.5.0)
Requirement already satisfied: gitdb<5,>=4.0.1 in ./miniconda/envs/py39/lib/python3.9/site-packages (from gitpython->mknotebooks->nxviz==0.7.3->dynamo-release) (4.0.9)
Requirement already satisfied: tornado>=4.1 in ./miniconda/envs/py39/lib/python3.9/site-packages (from jupyter-client->mknotebooks->nxviz==0.7.3->dynamo-release) (6.1)
Requirement already satisfied: nest-asyncio>=1.5 in ./miniconda/envs/py39/lib/python3.9/site-packages (from jupyter-client->mknotebooks->nxviz==0.7.3->dynamo-release) (1.5.4)
Requirement already satisfied: pyzmq>=13 in ./miniconda/envs/py39/lib/python3.9/site-packages (from jupyter-client->mknotebooks->nxviz==0.7.3->dynamo-release) (22.3.0)
Requirement already satisfied: smmap<6,>=3.0.1 in ./miniconda/envs/py39/lib/python3.9/site-packages (from gitdb<5,>=4.0.1->gitpython->mknotebooks->nxviz==0.7.3->dynamo-release) (5.0.0)
Requirement already satisfied: jsonschema!=2.5.0,>=2.4 in ./miniconda/envs/py39/lib/python3.9/site-packages (from nbformat>=4.4->nbconvert>=6.0.0->mknotebooks->nxviz==0.7.3->dynamo-release) (4.4.0)
Requirement already satisfied: ipython-genutils in ./miniconda/envs/py39/lib/python3.9/site-packages (from nbformat>=4.4->nbconvert>=6.0.0->mknotebooks->nxviz==0.7.3->dynamo-release) (0.2.0)
Requirement already satisfied: webencodings in ./miniconda/envs/py39/lib/python3.9/site-packages (from bleach->nbconvert>=6.0.0->mknotebooks->nxviz==0.7.3->dynamo-release) (0.5.1)
Requirement already satisfied: attrs>=17.4.0 in ./miniconda/envs/py39/lib/python3.9/site-packages (from jsonschema!=2.5.0,>=2.4->nbformat>=4.4->nbconvert>=6.0.0->mknotebooks->nxviz==0.7.3->dynamo-release) (21.4.0)
Requirement already satisfied: pyrsistent!=0.17.0,!=0.17.1,!=0.17.2,>=0.14.0 in ./miniconda/envs/py39/lib/python3.9/site-packages (from jsonschema!=2.5.0,>=2.4->nbformat>=4.4->nbconvert>=6.0.0->mknotebooks->nxviz==0.7.3->dynamo-release) (0.18.1)
```
Then I try to import dynamo in python3.9
```
>>> import dynamo as dyn
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "/home/liangyue/.local/lib/python3.9/site-packages/dynamo_release-1.0.9-py3.9.egg/dynamo/__init__.py", line 6, in <module>
    __version__ = get_version(__file__)
  File "/home/liangyue/.local/lib/python3.9/site-packages/get_version-3.5.4-py3.9.egg/get_version/__init__.py", line 280, in get_version
    raise NoVersionFound(Source.all, msg)
get_version.NoVersionFound: No version found:
- Directory name: name of directory “/home/liangyue/.local/lib/python3.9/site-packages/dynamo_release-1.0.9-py3.9.egg” does not contain a valid version.
- VCS: could not find VCS from directory “/home/liangyue/.local/lib/python3.9/site-packages/dynamo_release-1.0.9-py3.9.egg”.
- Package metadata: could not find distribution “dynamo”.
```

**Comment by Xiaojieqiu** [comment]
you need to upgrade to 1.1.0 via `pip install -U dynamo-release`

**Comment by FionaMoon** [comment]
Alright,thank you.



| |
***@***.***
|
|
***@***.***
|




---- Replied Message ----
| From | Xiaojie ***@***.***> |
| Date | 05/17/2022 12:55 |
| To | ***@***.***> |
| Cc | ***@***.******@***.***> |
| Subject | Re: [aristoteleo/dynamo-release] Problem with installation (Issue #327) |

Closed #327.
